### PR TITLE
Dict utils

### PR DESCRIPTION
**Pull request by ilteoood**

### CODE REVIEW

File: bin/actions/ActionInterface.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  void executeAction(final List<String> params);
+
+  Future<dynamic> loadPubspec() async {
+    final String pubSpecContent = await File("./pubspec.yaml").readAsString();
+    return loadYaml(pubSpecContent);
+  }
+
+  Future<List<String>> retrieveAssetsFolders() async {
+    dynamic pubSec = await loadPubspec();
```
Comment (defint) [inline bin/actions/ActionInterface.dart:18]: Seems like YamlMap can be used here.

File: bin/actions/ActionInterface.dart [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import 'dart:io';
+
+import 'package:yaml/yaml.dart';
+import 'package:path/path.dart';
+
+abstract class AbstractAction {
+
+  List<String> get acceptedExtensions => ['.json', '.yaml', '.xml'];
```
Comment (defint) [inline bin/actions/ActionInterface.dart:8]: Feature request: possibility to ignore specific files or folders.
For example, I have some .json files to load mock data.

File: bin/utils/LocalLoader.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  LocalLoader(this.fileSystemEntity);
+
+  @override
+  Future<String> loadString(final String fileName, final String extension) {
+    return File(this.fileSystemEntity.path).readAsString();
+  }
+
+  Future<Map> loadContent() async {
+    final BaseDecodeStrategy decodeStrategy = findStrategy(fileSystemEntity);
+    return await decodeStrategy.decode(
```
Comment (defint) [inline bin/utils/LocalLoader.dart:22]: Unnecessary await.

File: bin/actions/DiffAction.dart [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    final String compareFile = params[1];
+    final List<FileSystemEntity> assetsContent = await retrieveAssetsContent();
+    final Map baseFileContent =
+        await retrieveFileContent(assetsContent, baseFile);
+    final Map compareFileContent =
+        await retrieveFileContent(assetsContent, compareFile);
+    mapCompare(baseFileContent, compareFileContent, "");
+  }
+
+  mapCompare(final Map baseFileContent, final Map compareFileContent,
```
Comment (defint) [inline bin/actions/DiffAction.dart:22]: Did you check the package https://github.com/google/dart-json_diff ? 
Maybe ca reduce some amount of code.

**Review by defint — APPROVED** [review]

**Comment by defint** [comment]
Nice work!
It opens possibilities to add some useful features.

File: bin/actions/DiffAction.dart (hunk 4 above)
Comment (ilteoood) [reply bin/actions/DiffAction.dart:22]: AFAIK that library isn't published on pub anymore: its latest version is 0.1.2...and packages with git dependencies can't be published

**Pull request closed without merging**